parsec/core/fs/storage/manifest_storage.py

```diff
@@ -88,6 +92,47 @@ async def _create_db(self):
                 );
                 """
             )
+            # Singleton storing the pattern_filter
+            cursor.execute(
+                """
+                CREATE TABLE IF NOT EXISTS pattern_filter
+                (
+                  _id INTEGER PRIMARY KEY NOT NULL,
+                  pattern TEXT NOT NULL,
+                  fully_applied INTEGER NOT NULL  -- Boolean
+                );
+                """
+            )
+            # Set the default pattern filter if it doesn't exist
+            cursor.execute(
+                """
+                INSERT OR IGNORE INTO pattern_filter(_id, pattern, fully_applied)
+                VALUES (0, ?, 0)""",
+                (EMPTY_PATTERN,),
+            )
+
+    # Pattern filter operations
+
+    async def get_pattern_filter(self) -> Tuple[Pattern, bool]:
+        async with self._open_cursor() as cursor:
+            cursor.execute("SELECT pattern, fully_applied FROM pattern_filter WHERE _id = 0")
+            reply = cursor.fetchone()
+            pattern, fully_applied = reply
+            return (re.compile(pattern), bool(fully_applied))
+
+    async def set_pattern_filter(self, pattern: Pattern):
+        async with self._open_cursor() as cursor:
+            cursor.execute(
+                """UPDATE pattern_filter SET pattern = ?, fully_applied = 0 WHERE _id = 0 AND pattern != ?""",
```

Comment:
what the need for the `AND pattern != ?` ? Is it because we can avoid a more costly write this way (and `pattern == ?` is often true)